pkg/operators/formatters/formatters.go

```diff
@@ -323,24 +337,68 @@ var replacers = []replacer{
 			if ports[0].Size() != 2 {
 				return nil, fmt.Errorf("port size expected to be 2 bytes")
 			}
-			l3 := in.GetSubFieldsWithTag("type:" + L3EndpointTypeName)
-			if len(l3) != 1 {
-				return nil, fmt.Errorf("expected exactly 1 l3endpoint field")
+			ips := in.GetSubFieldsWithTag("type:" + IPAddrTypeName)
+			if len(ips) != 1 {
+				return nil, fmt.Errorf("expected %d %q field, got %d", 1, IPAddrTypeName, len(ips))
 			}
-			l3strings := l3[0].GetSubFieldsWithTag("l3string")
-			if len(l3strings) != 1 {
-				return nil, fmt.Errorf("expected exactly 1 l3string field")
+			if ips[0].Size() != 16 {
+				return nil, fmt.Errorf("expected %q to have 16 bytes, got %d", IPAddrTypeName, ips[0].Size())
 			}
+			versions := in.GetSubFieldsWithTag("name:version")
+			if len(versions) != 1 {
+				return nil, fmt.Errorf("expected exactly 1 version field")
+			}
+
+			// Hide all subfields
+			in.SetHidden(true, true)
+			// Show only root field
+			in.SetHidden(false, false)
 
-			// Hide l3 & subfields of l3
-			l3[0].SetHidden(true, true)
-			out, err := in.AddSubField("address", api.Kind_String)
+			// TODO: hidden and json.SkipFieldAnnotation instead?
+			ips[0].RemoveReference(true)
+
+			// Pretty L3 address
+			addrName := strings.TrimSuffix(ips[0].Name(), "_raw")
+			addrF, err := in.AddSubField(addrName, api.Kind_String,
+				datasource.WithFlags(datasource.FieldFlagHidden),
+			)
+			if err != nil {
+				return nil, fmt.Errorf("adding address field: %w", err)
+			}
```

Comment:
Is it possible to generalize this part from `l3endpoint` to only write this code once?

Reply:
I wasn't able to make it work last time, but I tried again and was able to create something. Would you mind checking it again?

Reply:
Looks good now, thanks